ferretdb/ferretdb.go

```diff
@@ -80,13 +84,13 @@ func New(config *Config) (*FerretDB, error) {
 	version.Get().Package = "embedded"
 
 	sp, err := state.NewProviderDir(config.StateDir)
-	if err == nil {
-		// TODO https://github.com/FerretDB/FerretDB/issues/4750
-		err = sp.Update(func(s *state.State) {
-			s.TelemetryLocked = true
-		})
+	if err != nil {
+		return nil, fmt.Errorf("failed to set up state provider: %w", err)
 	}
-
+	// TODO https://github.com/FerretDB/FerretDB/issues/4750
+	err = sp.Update(func(s *state.State) {
+		s.LatestVersion = version.Get().Version
+	})
```

Comment:
This doesn't look like it belongs here. The `LatestVersion` is reported by telemetry.

```suggestion
```

Reply:
Okay, can you give me some suggestions on what fields should be updated in the this function?

Reply:
There is no field you need to update :hugs: You've set the telemetry to be configurable in embedded package, so `TelemetryLocked` no longer need to be updated.